src/main/java/com/avispl/dal/avdevices/audionetworkinterface/studiotechnologies/danteleaderclock5401a/DanteLeaderClockCommunicator.java

```diff
@@ -109,61 +119,82 @@ public List<Statistics> getMultipleStatistics() throws Exception {
 			logger.debug(String.format("Perform getMultipleStatistics() at host: %s, port: %s", this.getHost(), this.getPort()));
 		}
 		final Map<String, String> stats = new HashMap<>();
-
-		failedMonitor.clear();
-		generalProperties(stats);
-		syncInputProperties(stats);
-		networkProperties(stats);
-		toneGeneratorProperties(stats);
-		systemProperties(stats);
-		if (failedMonitor.size() == DanteLeaderClockConstant.NO_OF_MONITORING_METRICS) {
-			StringBuilder stringBuilder = new StringBuilder();
-			stringBuilder.append(DanteLeaderClockConstant.FAIL_TO_GET_MONITORING_DATA);
-			for (Map.Entry<String, String> messageFailed : failedMonitor.entrySet()) {
-				String value = messageFailed.getValue();
-				if (value.contains(DanteLeaderClockConstant.FAIL_POPULATE_ERROR_MESSAGE)) {
-					continue;
-				}
-				stringBuilder.append(value);
-			}
-			failedMonitor.clear();
-			throw new ResourceNotReachableException(stringBuilder.toString());
+		if (localExtendedStatistics == null) {
+			localExtendedStatistics = new ExtendedStatistics();
+			localExtendedStatistics.setStatistics(stats);
+		}
+		if (sentRequests.get(DanteLeaderClockConstant.NETWORK).equals(false)) {
```

Comment:
according to this logic, it's gonna send all the requests one immediately after the other, since there are no other checks, except for if it has been retrieved already or not. I may be getting this wrong, so please explain if you will :) 

I was under impression that what we need is "1st cycle -> request 1 -> 2nd cycle -> request 3 -> 3rd cycle -> request 3 -> 4th cycle -> request 1 -> .......", but I may be missing some context